pkg/controller/llmisvc/webhook/llminferenceservice_validator.go

```diff
@@ -171,3 +170,97 @@ func (l *LLMInferenceServiceValidator) validateRouterCrossFieldConstraints(llmSv
+
+	allErrs = append(allErrs, l.validateWorkloadParallelism(field.NewPath("spec"), &llmSvc.Spec.WorkloadSpec)...)
+
+	if llmSvc.Spec.Prefill != nil {
+		allErrs = append(allErrs, l.validateWorkloadParallelism(field.NewPath("spec").Child("prefill"), llmSvc.Spec.Prefill)...)
+	}
+
+	return allErrs
+}
+
+func (l *LLMInferenceServiceValidator) validateWorkloadParallelism(basePath *field.Path, workload *v1alpha1.WorkloadSpec) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if workload.Worker != nil && workload.Parallelism == nil {
+		allErrs = append(allErrs, field.Invalid(
+			basePath.Child("worker"),
+			workload.Worker,
+			"when worker is specified, parallelism must be configured for either data parallelism or pipeline parallelism",
+		))
+		return allErrs
+	}
+
+	if workload.Parallelism == nil {
+		return field.ErrorList{}
+	}
+
+	parallelismPath := basePath.Child("parallelism")
+	parallelism := workload.Parallelism
+
+	// setting both Pipeline and any of DataLocal or Data is rejected
+	if parallelism.Pipeline != nil && (parallelism.Data != nil || parallelism.DataLocal != nil) {
+		allErrs = append(allErrs, field.Invalid(
+			parallelismPath,
+			parallelism,
+			"cannot set both pipeline parallelism and data parallelism (data or dataLocal) simultaneously",
+		))
+	}
+
+	// if Data is set, DataLocal must be set
+	if (parallelism.Data != nil) != (parallelism.DataLocal != nil) {
+		if parallelism.Data != nil && parallelism.DataLocal == nil {
+			allErrs = append(allErrs, field.Invalid(
+				parallelismPath.Child("dataLocal"),
+				parallelism.DataLocal,
+				"dataLocal must be set when data is set",
+			))
+		}
+		if parallelism.DataLocal != nil && parallelism.Data == nil {
+			allErrs = append(allErrs, field.Invalid(
+				parallelismPath.Child("data"),
+				parallelism.Data,
+				"data must be set when dataLocal is set",
+			))
+		}
+	}
+
+	if parallelism.Pipeline != nil && *parallelism.Pipeline <= 0 {
+		allErrs = append(allErrs, field.Invalid(
+			parallelismPath.Child("pipeline"),
+			*parallelism.Pipeline,
+			"pipeline parallelism must be greater than 0",
+		))
+	}
+
+	if parallelism.Data != nil && *parallelism.Data <= 0 {
+		allErrs = append(allErrs, field.Invalid(
+			parallelismPath.Child("data"),
+			*parallelism.Data,
+			"data parallelism must be greater than 0",
+		))
+	}
+
+	if parallelism.DataLocal != nil && *parallelism.DataLocal <= 0 {
+		allErrs = append(allErrs, field.Invalid(
+			parallelismPath.Child("dataLocal"),
+			*parallelism.DataLocal,
+			"dataLocal parallelism must be greater than 0",
+		))
+	}
+
+	// when the worker is defined, one of IsDataParallel() or IsPipelineParallel() must be used
+	if workload.Worker != nil && !parallelism.IsDataParallel() && !parallelism.IsPipelineParallel() {
+		allErrs = append(allErrs, field.Invalid(
+			basePath.Child("worker"),
+			workload.Worker,
+			"when worker is specified, parallelism must be configured for either data parallelism or pipeline parallelism",
+		))
+	}
```

Comment:
nit: I'd place this check co-located with the check at line 206 (`setting both Pipeline and any of DataLocal or Data is rejected`)